src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -49,12 +50,18 @@ export default function getAdaptationSwitchStrategy(
   queuedSourceBuffer : QueuedSourceBuffer<unknown>,
   period : Period,
   adaptation : Adaptation,
-  clockTick : { currentTime : number; readyState : number }
+  clockTick : { currentTime : number; readyState : number },
+  audioTrackSwitchingMode: "smooth" | "flush" | "reload"
 ) : IAdaptationSwitchStrategy {
   const buffered = queuedSourceBuffer.getBufferedRanges();
   if (buffered.length === 0) {
     return { type: "continue", value: undefined };
   }
+
+  if (adaptation.type === "audio" && audioTrackSwitchingMode === "reload") {
```

Comment:
Wouldn't that mean reload at every adaptation change (even the first)?

I first thought that we could simply take the same conditions than for video (lower in that function), but then realized that we might have codecs issues when not reloading - because for example no segment from the old Adaptation has been pushed around the current time - yet still switching the audio track (this should also be a problem with video by the way, so perhaps we have just found an issue there).

You might be tempted to go back to the situation where you added a boolean to that function based on the index of the original observable, but I don't think that this is a good idea.
I'm not a fan of using the index argument provided by observables there, because it implies the false assumption that the first emitted adaptation in a given `AdaptationStream` is the first ever chosen track for that Period (I do not even think that this is always true, for example we might have issues when seeking back to a previous Period, as it would be the first choice for the new `AdaptationStream` we just created for that Period, but an old, since removed, `AdaptationStream` had already pushed segments there. So we wouldn't be reloading despite needing to do it).

So I would much prefer if we are very specific about what we want, not just rely on assumptions.
Maybe we could do both what we're doing for video but also checking if the codec is the same (we can interrogate the `queuedSourceBuffer` here for that)?

Reply:
Ok, I just removed the last try (commit). I feel a little lost, it would be nice if we can speak together to make things clear because it seems that there is potentially a lot of side effect possible and to make sure we are going in the right direction.